### PR TITLE
feat(footer): add footer shadow parts

### PR DESCRIPTION
**Pull request by bruno-amorim**
### Related Ticket(s)

[ADCMS-5147](https://jsw.ibm.com/browse/ADCMS-5147)

### Description

Adding shadow parts to the footer component to allow light customizations to component internals.
Snapshots updated.

### CODE REVIEW

**Review by jkaeser — COMMENTED** [review]
Looking good to me, but the ci-checks are failing. Can you take a look, @bruno-amorim?

**Review by jkaeser — APPROVED** [review]

**Review by andy-blum — CHANGES_REQUESTED** [review]
There are some component files that don't seem to have the necessary part attributes:
* legal-nav is missing some elements
* combo-box doesn't seem to have any part attributes

File: packages/web-components/src/components/footer/footer-nav-group.ts [hunk 1]
```diff
@@ -127,14 +132,17 @@ class C4DFooterNavGroup extends MediaQueryMixin(
...
           </button>
         `;
     return html`
       ${heading}
-      <div id="content" class="${prefix}--accordion__content">
+      <div
+        part="accordion-content"
+        id="content"
+        class="${prefix}--accordion__content">
         <ul>
```
Comment (m4olivei) [inline packages/web-components/src/components/footer/footer-nav-group.ts:146]: Missing a `part` attribute here.

File: packages/web-components/src/components/footer/legal-nav.ts [hunk 2]
```diff
@@ -76,15 +79,18 @@ class C4DLegalNav extends StableSelectorMixin(LitElement) {
     return this.size !== FOOTER_SIZE.MICRO
       ? html`
           <nav
+            part="legal-nav"
             class="${c4dPrefix}--legal-nav"
             aria-label="${ifDefined(navLabel)}">
-            <div class="${this._getTypeClass()}">
+            <div part="legal-nav-list" class="${this._getTypeClass()}">
               <ul>
```
Comment (m4olivei) [inline packages/web-components/src/components/footer/legal-nav.ts:86]: Missing a `part` attribute here. I'd suggest naming the above `<div>` on line 85 as `legal-nav-list-container` and this`<ul>` take the name `legal-nav-list`.

File: packages/web-components/src/components/footer/legal-nav.ts [hunk 3]
```diff
@@ -76,15 +79,18 @@ class C4DLegalNav extends StableSelectorMixin(LitElement) {
...
                 <slot></slot>
               </ul>
               <slot name="locale"></slot>
             </div>
-            <div class="${c4dPrefix}--adjunct-links__container">
+            <div
+              part="adjunct-links-container"
+              class="${c4dPrefix}--adjunct-links__container">
               <ul>
```
Comment (m4olivei) [inline packages/web-components/src/components/footer/legal-nav.ts:94]: Missing a `part` attribute here. Recommend `adjunct-links-list`.

Also, review lines 100-111 below. Use the same `legal-nav*` part names as above lines 81-98. If you look at the Footer > Micro story, you can see how these are presently missing.

**Review by m4olivei — CHANGES_REQUESTED** [review]
Few that I found were missing still.

Not part of the current diff, but I also found one missing in the = `<c4d-legal-nav-cookie-preferences-placeholder>` element:

[c4d-legal-nav-cookie-preferences-placeholder](https://github.com/carbon-design-system/carbon-for-ibm-dotcom/blob/feat/footer-shadow-parts/packages/web-components/src/components/footer/legal-nav-cookie-preferences-placeholder.ts#L55)

**Review by m4olivei — APPROVED** [review]

**Pull request merged**